data/pools/transactionPool.go

```diff
@@ -435,6 +301,92 @@ func (pool *TransactionPool) OnNewBlock(block bookkeeping.Block) {
 	}
 }
 
+// alwaysVerifiedPool implements ledger.VerifiedTxnCache and returns every
+// transaction as verified.
+type alwaysVerifiedPool struct{}
+
+func (*alwaysVerifiedPool) Verified(txn transactions.SignedTxn) bool {
+	return true
+}
+
+func (pool *TransactionPool) addToPendingBlockEvaluatorOnce(tx *transactions.SignedTxn) error {
+	if tx.LastValid < pool.pendingBlockEvaluatorRound {
+		return fmt.Errorf("Transaction valid for %d..%d which is before %d", tx.FirstValid, tx.LastValid, simulatedNextRound)
+	}
+
+	return pool.pendingBlockEvaluator.Transaction(tx, nil)
+}
+
+func (pool *TransactionPool) addToPendingBlockEvaluator(tx *transactions.SignedTxn) error {
+	err := pool.addToPendingBlockEvaluatorOnce(tx)
+	if err == ledger.ErrNoSpace {
+		pool.pendingBlockEvaluatorRound++
+		pool.pendingBlockEvaluator.ResetTxnBytes()
+		err = pool.addToPendingBlockEvaluatorOnce(tx)
+	}
+	return err
+}
+
+// recomputeBlockEvaluator constructs a new BlockEvaluator and feeds all
+// in-pool transactions to it (removing any transactions that are rejected
+// by the BlockEvaluator).
+func (pool *TransactionPool) recomputeBlockEvaluator() (stats telemetryspec.ProcessBlockMetrics) {
+	pool.pendingBlockEvaluator = nil
+
+	latest := pool.ledger.Latest()
+	prev, err := pool.ledger.BlockHdr(latest)
+	if err != nil {
+		logging.Base().Warnf("TransactionPool.recomputeBlockEvaluator: cannot get prev header for %d: %v",
+			latest, err)
+		return
+	}
+
+	next := bookkeeping.MakeBlock(prev)
+	pool.pendingBlockEvaluatorRound = next.Round()
+	pool.pendingBlockEvaluator, err = pool.ledger.StartEvaluator(next.BlockHeader, &alwaysVerifiedPool{}, nil)
+	if err != nil {
+		logging.Base().Warnf("TransactionPool.recomputeBlockEvaluator: cannot start evaluator: %v", err)
```

Comment:
Likewise - should this be an error?

Reply:
Similarly, can legitimately happen during catchup.